Added the README's for cross-contract-calls & factory-contract.

MattCS2006 opened the pull request:
This pull request is for the issue #152.

I made the README's for both folders, **cross-contract-calls** and **factory-contract** following the standard used on the other folders that had a README within the folder of examples.

MattCS2006 committed:
Added the README's for cross-contract-calls & factory-contract.

I made the README's for both folders following the standard used on the other folders that had a README within the folder of examples.

frol commented on examples/cross-contract-calls/README.md:
````diff
@@ -1,5 +1,58 @@
+
+### Usage
+
+```markdown
````
What is the benefit of having this block in ```markdown?

MattCS2006 replied:
There is no benefit other than aesthetic honestly, I can change it if you want! Just let me know @frol

frol commented on examples/cross-contract-calls/README.md:
````diff
@@ -1,5 +1,58 @@
+```markdown
+## Usage
+
+1. Deploy the contracts:
+
+   - Deploy the high-level contract.
+   - Deploy the low-level contract.
+
+2. Call the high-level contract to initiate a cross-contract call to the low-level contract.
````
Please, document how to do that with `cargo near deploy`:

1. Prerequisite: [install `cargo-near`](https://github.com/near/cargo-near)
2. (optional) Create a new testnet account: `cargo near create-dev-account`
3. Build and deploy the contract: `cargo near deploy` + add a screenshot here
4. Interact with the contract from [CLI](https://near.cli.rs): `near contract call-function ...` + add a screenshot here

frol replied:
Just for the reference, I have tested it here:

<img width="1151" alt="image" src="https://github.com/user-attachments/assets/860e2cf5-39b2-4441-b1b5-4127088af908">

P.S. There is a small bug in near.cli.rs, so you have to change the `pub fn factorial(&self` into `pub fn factorial(&mut self`

MattCS2006 replied:
Got it @frol, will do as soon as possible!

MattCS2006 replied:
Just made the changes

frol reviewed (changes requested):
Please, address my review comments (and apply them to both READMEs)

MattCS2006 committed:
Merge branch 'master' into Write-a-README-for-factory-contract-and-cross-contract-examples

MattCS2006 committed:
Changes on the usage and the markdown

Just made the changes for the usage part and gave it more details on how to do it, like an step-by-step, and also got rid of unnecessary markdown

frol commented on examples/factory-contract/README.md:
````diff
@@ -1,3 +1,75 @@
+3. Deploy the factory contract:
+
+```bash
+cargo near deploy --wasmFile path/to/factory-contract.wasm --accountId your-account.testnet
````
This command is invalid. Please, test all the commands

```suggestion
cargo near deploy
```

frol commented on examples/factory-contract/README.md:
````diff
@@ -1,3 +1,75 @@
+4. Use the factory contract to create instances of the target contract:
+
+```bash
+near call your-factory-contract-account.testnet create_instance '{"args": "value"}' --accountId your-account.testnet
````
This command is outdated, please, use the new syntax (it is suggested to you in the output):

```suggestion
near contract call-function as-transaction your-factory-contract-account.testnet deploy_status_message json-args '{"account_id": "sub.your-factory-contract-account.testnet"}' prepaid-gas '100 Tgas' attached-deposit '0 NEAR'
```

frol commented on examples/factory-contract/README.md:
````diff
@@ -1,3 +1,75 @@
+```
+
+NOTE: Replace `path/to/factory-contract.wasm` with the actual path to your compiled WebAssembly file, and `your-account.testnet` with your actual NEAR testnet account ID.
+
````
Demonstrate calls to `simple_call` and `complex_call` functions.

frol commented on examples/cross-contract-calls/README.md:
````diff
@@ -1,5 +1,85 @@
+- Call the high-level contract to initiate a cross-contract call to the low-level contract:
+
+```bash
+near call your-high-level-contract-account.testnet call-function '{"args": "value"}' --accountId your-account.testnet
````
This does not demonstrate the actual function defined in the contract (`factorial`). Please, write the example to the specific contract at hand, not just some placeholder

MattCS2006 committed:
Merge branch 'master' into Write-a-README-for-factory-contract-and-cross-contract-examples

MattCS2006 committed:
docs: Update README for factory contract and cross-contract examples with the changes requested.

MattCS2006 commented:
I just updated the README's @frol, please review them, thanks

r-near commented:
Closing due to 19+ months without addressing review feedback (conflicts with master). Documentation contributions still welcome — please open a fresh PR rebased on master if you'd like to revive this.
